vllm/multimodal/image.py

```diff
@@ -32,15 +42,21 @@ def __init__(self, *, image_mode: str = "RGB") -> None:
     def load_bytes(self, data: bytes) -> Image.Image:
         image = Image.open(BytesIO(data))
         image.load()
-        return image.convert(self.image_mode)
+        if image.mode == "RGBA" and self.image_mode == "RGB":
+            return rgba_to_rgb(image)
+        else:
+            return image.convert(self.image_mode)
```

Comment:
Maybe you could just make a image_to_image_mode function that has this conditional inside of it